### PR TITLE
Add jonnys redux changes

### PR DESCRIPTION
**Pull request by stevenjob**

### CODE REVIEW

**Comment by jdhodges** [comment]
LGTM :+9999:

**Comment by jdhodges** [comment]
Nice work. Merged :+1:

File: frontend/src/components/ChatList.jsx [hunk 1]
```diff
@@ -1,20 +1,21 @@
 import React from 'react';
 import ChatComment from './ChatComment.jsx';
 
-const ChatList = (props) => {
-  console.log(this);
-  let commentNodes = props.data.map((comment) =>
+const ChatList = ({ comments }) => {
```
Comment (rjmcneill) [inline frontend/src/components/ChatList.jsx:4]: don't need `( )` around this

File: frontend/src/components/ChatList.jsx [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+  let commentNodes = comments.map((comment) =>
```
Comment (rjmcneill) [inline frontend/src/components/ChatList.jsx:5]: don't need `( )` around this

File: frontend/src/components/ChatWindow.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { addComment } from '../actions/index.jsx';
+import io from 'socket.io-client';
+
+let dispatch;
+const socket = io.connect('http://localhost:9092');
+
+class ChatWindow extends React.Component {
+  componentDidMount() {
+    dispatch = this.props.dispatch;
+    socket.on('chatevent', (data) => {
```
Comment (rjmcneill) [inline frontend/src/components/ChatWindow.jsx:14]: don't need `( )` around this

File: frontend/src/components/HeaderBar.jsx [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import React from 'react';
 
 const HeaderBar = () => (
```
Comment (rjmcneill) [inline frontend/src/components/HeaderBar.jsx:3]: Will this work?

`const HeaderBar => {(
)}`

File: frontend/src/components/ChatList.jsx (hunk 1 above)
Comment (jdhodges) [inline frontend/src/components/ChatList.jsx:4]: we should update the linter pls

Comment (rjmcneill) [inline frontend/src/components/ChatList.jsx:4]: You're right Jonny, thanks for the suggestion.
We should definitely do that

**Pull request merged**